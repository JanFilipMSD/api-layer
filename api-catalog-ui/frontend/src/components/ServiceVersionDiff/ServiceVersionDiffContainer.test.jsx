/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ServiceVersionDiffContainer from './ServiceVersionDiffContainer';

const mockStore = configureStore();

describe('ServiceVersionDiff Container', () => {
    let store;
    let container;
    beforeEach(() => {
        store = mockStore({
            serviceVersionDiff: {
                diffText: 'diffText',
                oldVersion: '1.0',
                newVersion: '2.0',
            },
        });
        container = render(
            <Provider store={store}>
                <ServiceVersionDiffContainer versions={['v1', 'v2']} />
            </Provider>
        );
    });

    it('should render the container', () => {
        expect(container).not.toBeNull();
    });
});
