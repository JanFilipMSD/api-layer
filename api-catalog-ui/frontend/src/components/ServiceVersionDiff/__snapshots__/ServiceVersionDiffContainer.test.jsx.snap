// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceTab Container should render the container 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="api-diff-container"
      >
        <div
          class="api-diff-form"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
            data-testid="compare-label"
          >
            Compare
          </p>
          <div
            class="MuiFormControl-root formField"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
            >
              Version
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              data-testid="select-1"
              sx="[object Object]"
            >
              <div
                aria-haspopup="listbox"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                tabindex="-1"
                value="[object Object]"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1"
            data-testid="label-with"
          >
            with
          </p>
          <div
            class="MuiFormControl-root formField"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
            >
              Version
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              data-testid="select-2"
              sx="[object Object]"
            >
              <div
                aria-haspopup="listbox"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                tabindex="-1"
                value="[object Object]"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            data-testid="diff-button"
            id="diff-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              Go
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="api-diff-content"
        >
          diffText
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="api-diff-container"
    >
      <div
        class="api-diff-form"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
          data-testid="compare-label"
        >
          Compare
        </p>
        <div
          class="MuiFormControl-root formField"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
          >
            Version
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            data-testid="select-1"
            sx="[object Object]"
          >
            <div
              aria-haspopup="listbox"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="[object Object]"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1"
          data-testid="label-with"
        >
          with
        </p>
        <div
          class="MuiFormControl-root formField"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
          >
            Version
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            data-testid="select-2"
            sx="[object Object]"
          >
            <div
              aria-haspopup="listbox"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="[object Object]"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          data-testid="diff-button"
          id="diff-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            Go
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="api-diff-content"
      >
        diffText
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
